src/java/org/apache/cassandra/repair/ValidationTask.java

```diff
@@ -70,56 +72,50 @@ public synchronized void treesReceived(MerkleTrees trees)
     {
         if (trees == null)
         {
-            active = false;
             tryFailure(RepairException.warn(desc, previewKind, "Validation failed in " + endpoint));
         }
-        else if (active)
-        {
-            trySuccess(new TreeResponse(endpoint, trees));
-        }
-        else
+        else if (!trySuccess(new TreeResponse(endpoint, trees)))
         {
-            // If the task has already been aborted, just release the possibly off-heap trees and move along.
+            // If the task is done, just release the possibly off-heap trees and move along.
             trees.release();
-            trySuccess(null);
         }
     }
 
+    public synchronized void abort()
```

Comment:
Does this need to be `synchronized` when it's just calling the other `abort(Throwable)`, which is also `synchronized`?